examples/smalltalk/README.md

```diff
@@ -1,26 +1,17 @@
 # GNU Smalltalk
-
-## The Smalltalk language
-
 Smalltalk is probably the most "classic object-oriented" programming language. There are many
 dialects and implementations out there, but this example uses 
 [GNU Smalltalk](https://www.gnu.org/software/smalltalk/).
 
 Smalltalk's syntax is quite different from most other languages, but it's also very simple
-and extremely consistent, so it doesn't take long to learn. 
+and extremely consistent, so it doesn't take long to learn. In fact, the entire language
+specification can be printed in readable text on a postcard.
 
-## How to build
-```bash
-docker build -t aoc_smalltalk .
-```
+Alan Kay's original vision for Objects was "computers all the way down", where each object was to be seen as a computer receiving messages from its environment. Smalltalk didn't quite reach that idea, perhaps, but it is still probably the purest OO language in existence.
 
-## How to run in Docker
-The environment variable "part" specifies which part of the solution to run. Please adjust your code accordingly (see example in ``main.st``).
-```bash
-docker run -e part=part1 aoc_smalltalk
-```
+One of the most distinguishing features of Smalltalk is that you would generally program in a live environment and your objects can be evolved to gain new capabilities as the program runs (Smalltalk is dynamically typed). Smalltalk is possibly the most productive language in existence, but probably lost the language wars because you had to pay for it.
```

Comment:
@robu är detta korrekt? Jag inser att jag kanske rört till det. Kan ett objekt "lära" sig tolka meddelanden som inte andra medlemmar av samma klass kan?

Reply:
"Live environment" köper jag in på, men resten är inget jag kan skriva under på. Jag kan dock inte säga att det är fel, bara att jag inte är smalltalk-expert, och skulle inte beskriva det på det sättet. Vi har nog andra ST-experter på bygget, om du vill bolla beskrivningen. Jonas Öhrn, kanske?

Reply:
Ja det stämmer att man programmerar live i Smalltalk med sin s.k. image som innehåller alla klasser för själva Smalltalkimplementationen, tex Gnu, och din egen kod.
Man kan dynamiskt ändra/skapa klasser även i runtime genom att göra "compile" på ändrad klass, dvs då kan du ändra vad som helst.
Att "lära sig tolka" tror jag härstämmar från doesNotUnderstand-mekanismen som beskrivs här:
https://wiki.c2.com/?DoesNotUnderstand


"...because you had to pay for it" 
skulle jag nog byta ut mot 
"...because the unusual syntax scared new developers so Smalltalk never reached critical mass in developer communities"